### ISSUE TITLE
No errors on array initialization if any element is of any type

### ISSUE THREAD

**Issue by chrisirhc** [microsoft/TypeScript#46190]
# Bug Report

### 🔎 Search Terms
* any type used in array initialization

### 🕗 Version & Regression Information

3.9.7 and beyond

### ⏯ Playground Link

https://www.typescriptlang.org/play?ts=3.9.7#code/MYewdgzgLgBBIFsCmYCuCBGSBOAuGamOMAvDAIwDcAUKJLPMlABYCWYA5kgDYRL4BDMAE9SFGnWgxg+aNnYcA2gF0xixinRZsAGjiIkLBTz7LKQA

### 💻 Code

```ts
const somenumber: number = 1;
const somethingelse: any = 1;
const c: string[] = [somenumber, somethingelse];
```

### 🙁 Actual behavior

No errors reported.

### 🙂 Expected behavior

Error reported on the array initialization, on line 3, on the use of `somenumber` in a string array .

### Side note
If this is expected behavior when any item is any, is there a way to have the typescript lint report this?
It seems odd that the initialization line reports no error or warning at all, since there's a specific type declared on that line.

**Comment by IllusionMH** [comment]
When you are using `any` you basically opt out of type checking  in that place and it's intentionally that there are no errors and can't be.

If you want to mark variable that can keep some value but you don't know what it is exact - use `unknown`.

**Comment by chrisirhc** [comment]
There was a mistake in my demo code/playground.
The issue isn't with the `any` variable, but with the `number` variable being unchecked.

I understand if `any` is not being checked, but I would've expected that the first element, `somenumber`, be checked.

**Comment by MartinJohns** [comment]
Duplicate of #19541. Used search terms: `array any in:title`

**Comment by chrisirhc** [comment]
Thanks folks. Closing since this is also an exact duplicate of #38870 , which was closed as dupe of #19541. Sorry, I didn't find that earlier.
